test-frame-common/src/main/java/io/skodjob/testframe/clients/KubeClient.java

```diff
@@ -98,6 +108,230 @@ public List<HasMetadata> readResourcesFromFile(InputStream is) throws IOExceptio
         }
     }
 
+    /**
+     * Check if namespace exists in current cluster
+     *
+     * @param namespace namespace name
+     * @return true if namespace exists
+     */
+    public boolean namespaceExists(String namespace) {
+        return client.namespaces().list().getItems().stream().map(n -> n.getMetadata().getName())
+                .toList().contains(namespace);
```

Comment:
One thing, which caught my eye...just an optimization because this implementation of namespaceExist could be very resource-intensive if one has a lot of namespaces (i.e., because of instantiation of the un-needed list)
```java
public boolean namespaceExists(String namespace) {
    return client.namespaces().list().getItems().stream()
                 .anyMatch(n -> n.getMetadata().getName().equals(namespace));
}
```

Reply:
thx!